docker/init.sh

```diff
@@ -0,0 +1,45 @@
+#! /bin/bash
+
+set -e
+
+# data directory is required for caching the secret key in a file
+if [ "$DATA_DIR" == "" ]; then
+    echo -n "Error: DATA_DIR environment variable not set!"
+    echo "Are you sure you are running this script in a Docker container?"
+    exit 1
+fi
+
+SECRET_KEY_FILE="$DATA_DIR/secret_key"
+
+if [ ! -f "$SECRET_KEY_FILE" ]; then
+    install -m 0600 /dev/null "$SECRET_KEY_FILE"
+    SECRET_KEY=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 100 | head -n 1)
+    echo "$key" > "$SECRET_KEY_FILE"
+    chmod -w "$SECRET_KEY_FILE"
+else
+    SECRET_KEY=$(cat "$SECRET_KEY_FILE")
+fi
+
+export SECRET_KEY
+export RAILS_ENV=production
+
+install -m 0600 /dev/null .my.cnf
+cat > .my.cnf <<ABC
+[client]
+user=$MYSQL_USER
+password=$MYSQL_PASSWORD
+ABC
+
+if [ $(echo "show tables;" | mysql --host $DATABASE_HOST --port $DATABASE_PORT $MYSQL_DATABASE | wc -l) -le 1 ]; then
+    echo ">>> Initializing database..."
+    dockerize -wait tcp://$DATABASE_HOST:$DATABASE_PORT -timeout 60s bundle exec rake db:schema:load
+fi
+
+echo ">>> Upgrading database..."
```

Comment:
Well, upgrading the database makes sure that whenever the OSEM repository is updated and a new container is built, changes to the schema will be applied to the database, too.
The only time the upgrades are applied unnecessarily is during the first run. This could be easily "fixed" by putting the upgrading part into an else clause of the preceding if block, but I think that's not really necessary. But if you insisted on it, I'd "fix" it.